plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordSetProvider.java

```diff
@@ -44,6 +48,31 @@ public JdbcRecordSetProvider(JdbcClient jdbcClient, @ForRecordCursor ExecutorSer
         this.executor = requireNonNull(executor, "executor is null");
     }
 
+    @Override
+    public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit split, ConnectorTableHandle table, List<? extends ColumnHandle> columns, DynamicFilter dynamicFilter)
+    {
+        if (isEnableDynamicFiltering(session)) {
+            long timeoutMillis = getDynamicFilteringWaitTimeout(session).toMillis();
```

Comment:
I find some weird behavior when trying this Postgres as a probe side. It almost seems random whether dynamic filtering is used or not.

I've set join_reordering_strategy='NONE' and also set dynamic_filtering_wait_timeout='10000ms'.

Executing the same join over and over I find some sometimes it uses dynamic filters and sometimes it does not.

In the debugger I see that a dynamic filter is passed here, but its tupledomain is "all".
I think this is outside of this code, but wondering if you have seen this too.
(The performance difference between the two cases is 250ms vs 38 seconds.)

Reply:
Seems we also need to implement lazy_dynamic_filters, so that the probe side can (optionally) wait for a bit for the dynamic filters to be ready.

Reply:
Update: This happens only with PARTITIONED joins.